src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -57,9 +57,16 @@ public static IEnumerable<string> GetResourceModels(ServiceDefinition serviceDef
                                                     .Where(modelName => !(IsBaseResourceModelName(modelName))
                                                                         && serviceDefinition.Definitions.ContainsKey(modelName)
                                                                         && IsAllOfOnModelNames(modelName, serviceDefinition.Definitions, xmsAzureResourceModels));
+
+            var unfilteredResourceCandidates = resourceModels.Union(modelsAllOfOnXmsAzureResources);
+
+            // Now filter all the resource models that are returned from a POST operation only 
```

Comment:
I don't see the "only" part reflected in the code but maybe I'm missing it.
Also, the indentation is misleading: the `.SelectMany` calls are perfectly aligned but are *not* operating on the same "level". I'd expect the second `SelectMany` to be on the same height as the inner `Where`, just break `pathObj => pathObj<HERE>.Where` and lines won't be that long.